GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -2945,9 +2946,26 @@ private void LaunchRebase(string command)
                 return;
             }
 
-            string rebaseCmd = GitCommandHelpers.RebaseCmd(
-                GetActualRevision(LatestSelectedRevision)?.FirstParentId?.ToString(), interactive: true, preserveMerges: false,
-                autosquash: false, autoStash: true, ignoreDate: false, committerDateIsAuthorDate: false, supportRebaseMerges: Module.GitVersion.SupportRebaseMerges);
+            IGitGpgController gpg = new GitGpgController(() => Module, new GPGSecretKeysParser());
+            bool commitSign = false;
+            string defaultKey = null;
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                commitSign = await gpg.GetCommitGPGSignAsync();
+                defaultKey = await gpg.GetDefaultKeyAsync();
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            });
+
+            string rebaseCmd = GitCommandHelpers.RebaseCmd(GetActualRevision(LatestSelectedRevision)?.FirstParentId?.ToString(),
```

Comment:
I think we better express this as:
```suggestion
            ThreadHelper.JoinableTaskFactory.Run(async () =>
            {
                < await switch to background thread >
                commitSign = await gpg.GetCommitGPGSignAsync();
                defaultKey = await gpg.GetDefaultKeyAsync();
            });

            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
            string rebaseCmd = GitCommandHelpers.RebaseCmd(GetActualRevision(LatestSelectedRevision)?.FirstParentId?.ToString(),
```